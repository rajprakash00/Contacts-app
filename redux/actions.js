//action types
export const UPDATE_USER = "UPDATE_USER";
export const UPDATE_CONTACT = "UPDATE_CONTACT";

//action creators

export const updateUser = (update) => ({
  type: UPDATE_USER,
  payload: update,
});

export const addContact = (newContact) => ({
  type: UPDATE_CONTACT,
  payload: newContact,
});
