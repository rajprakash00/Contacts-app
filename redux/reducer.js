import { combineReducers } from "redux";

import { UPDATE_USER, UPDATE_CONTACT } from "./actions";

const merge = (prev, next) => Object.assign({}, prev, next);

//reducers
const contactReducer = (state = [], action) => {
  if (action.type === UPDATE_CONTACT) return [...state, action.payload];
  return state;
};

const userReducer = (state = {}, action) => {
  switch (action.type) {
    case UPDATE_USER:
      return merge(state, action.payload);
    case UPDATE_CONTACT:
      return merge(state, { prevContact: action.payload });
    default:
      return state;
  }
};

const reducer = combineReducers({
  user: userReducer,
  contacts: contactReducer,
});

export default reducer;
