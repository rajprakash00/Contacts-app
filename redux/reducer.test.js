import reducer from "./reducer";
import * as actions from "./actions";

const DEFAULT_STATE = {
  user: {},
  contacts: [],
};

describe("contact reducer", () => {
  it("adds new user", () => {
    expect(
      reducer(
        DEFAULT_STATE,
        actions.addContact({
          name: "user A",
          phone: "1234567890",
        })
      )
    ).toMatchSnapshot();
  });
});

describe("user reducer", () => {
  it("updates user", () => {
    expect(
      reducer(
        DEFAULT_STATE,
        actions.updateUser({
          name: "user A",
        })
      )
    ).toMatchSnapshot();
  });
});
