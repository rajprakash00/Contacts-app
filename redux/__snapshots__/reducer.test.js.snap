// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact reducer adds new user 1`] = `
Object {
  "auth": Object {},
  "contacts": Array [
    Object {
      "name": "user A",
      "phone": "1234567890",
    },
  ],
  "user": Object {
    "prevContact": Object {
      "name": "user A",
      "phone": "1234567890",
    },
  },
}
`;

exports[`user reducer updates user 1`] = `
Object {
  "auth": Object {},
  "contacts": Array [],
  "user": Object {
    "name": "user A",
  },
}
`;
